### PR TITLE
feat(mobile): add categoriesStore for category management

### LINKED ISSUE #208: mobile: Create categoriesStore for category management
## Problem
No store exists for fetching/managing categories. Multiple screens need category data.

## Requirements
Create `mobile/src/stores/categoriesStore.ts` with:

### State
- `categories: Category[]`
- `isLoading: boolean`
- `error: string | null`

### Actions
- `fetchCategories()` - GET /api/v1/categories
- `createCategory(data)` - POST /api/v1/categories
- `archiveCategory(id)` - POST /api/v1/categories/[id]/archive
- `getCategoriesByType(type)` - filter INCOME/EXPENSE

### Types
Define `Category` type with id, name, type, color, isArchived

## Blocked by
- #195 (GET /api/v1/categories endpoint)

## Acceptance Criteria
- [ ] Store created with Zustand
- [ ] Fetch on app init
- [ ] Filter by type helper
- [ ] Unit tests added

### PR DESCRIPTION
**Pull request by avifenesh**
## Summary

Add Zustand store for category management in the mobile app:
- Create `categoriesStore` with fetchCategories, createCategory, archiveCategory, unarchiveCategory
- Move Category type definition to categoriesStore as authoritative source
- Update transactionsStore and budgetsStore imports
- Add comprehensive tests with edge cases

## Changes

- `mobile/src/stores/categoriesStore.ts` - New Zustand store
- `mobile/__tests__/stores/categoriesStore.test.ts` - Unit tests
- `mobile/src/stores/index.ts` - Export new store
- `mobile/src/stores/transactionsStore.ts` - Import Category from categoriesStore
- `mobile/src/stores/budgetsStore.ts` - Import Category from categoriesStore
- `mobile/README.md` - Document stores directory

## Test Plan

- [x] All mobile tests pass (`npm test` in mobile/)
- [x] TypeScript type checking passes
- [x] Store follows existing patterns (transactionsStore, budgetsStore)

## Related Issues

Closes #208

This is a foundation store that unblocks: #213, #214, #215, #216, #222

### CODE REVIEW

File: mobile/src/stores/categoriesStore.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+  createCategory: async (data: CreateCategoryInput) => {
+    const accessToken = useAuthStore.getState().accessToken;
+
+    try {
+      const category = await apiPost<Category>('/categories', data, accessToken);
+
+      set((state) => ({
+        categories: [...state.categories, category],
+      }));
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply mobile/src/stores/categoriesStore.ts]: Fixed! Updated createCategory to check if the returned category ID already exists in state. If it does (reactivation case), it updates the existing entry instead of appending a duplicate. Added test coverage for this scenario.

File: mobile/src/stores/categoriesStore.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        `/categories/${id}/archive`,
+        { isArchived: false },
+        accessToken
+      );
+
+      set((state) => ({
+        categories: state.categories.map((c) =>
+          c.id === id ? { ...c, isArchived: false } : c
+        ),
+      }));
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply mobile/src/stores/categoriesStore.ts]: Same explanation as above - the API only returns the minimal response. The unarchive operation is symmetrical to archive.

File: mobile/src/stores/categoriesStore.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+export type TransactionType = 'INCOME' | 'EXPENSE';
+
+export interface Category {
+  id: string;
+  name: string;
+  type: TransactionType;
+  color: string;
+  isArchived: boolean;
+  isHolding: boolean;
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply mobile/src/stores/categoriesStore.ts]: The fields are intentionally required, not optional. The API always returns these fields (see GET /categories endpoint response at src/app/api/v1/categories/route.ts:56-64). Making them optional would weaken type safety and require null checks throughout the codebase. All existing tests have been updated to include these fields.

File: mobile/src/lib/logger.ts [hunk 4]
```diff
@@ -1,9 +1,3 @@
-/**
- * Mobile logger for development debugging.
- * Only logs in development mode (__DEV__).
- * Sanitizes sensitive data before logging.
- */
-
 /* eslint-disable no-console -- This is the designated logging module */
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply mobile/src/lib/logger.ts]: The logger comment removal was done by the automated deslop-work agent during the pre-review cleanup phase. The comment was deemed unnecessary as the code is self-documenting (the eslint-disable comment already explains this is the designated logging module). This follows our CLAUDE.md guidelines about avoiding verbose comments that explain obvious code.
